instant-xml/src/lib.rs

```diff
@@ -10,12 +10,176 @@ pub use macros::{FromXml, ToXml};
 pub mod parse;
 
 pub trait ToXml {
-    fn write_xml<W: fmt::Write>(&self, write: &mut W) -> Result<(), Error>;
-
     fn to_xml(&self) -> Result<String, Error> {
-        let mut out = String::new();
-        self.write_xml(&mut out)?;
-        Ok(out)
+        let mut parent_prefixes = BTreeSet::new();
+        let mut output = String::new();
+        let mut serializer = Serializer {
+            parent_prefixes: &mut parent_prefixes,
+            output: &mut output,
+        };
+        self.serialize(&mut serializer, None)?;
+        Ok(output)
+    }
+
+    fn serialize(
+        &self,
+        serializer: &mut Serializer,
+        field_data: Option<&mut FieldData>,
```

Comment:
The idea is that you just modify the field_data if you want to customize the tag name, without the creating the new value.
In case of the custom type, by default it is taking the struct type as the xml tag (here)[https://github.com/InstantDomain/instant-xml/blob/349af40add28a0bf12142055355f3d1c754f32ec/instant-xml-macros/src/lib.rs#L219]
But if you would like to overide the serialize fn for some struct, you could just modify the `name` inside `FieldData` .

Reply:
Who would (or, where would you) modify the `field_data`? I'm not sure this idea makes sense. The `field_data` should be exclusively used by a parent type to pass some context into a child type.

Reply:
Like this:
`
#[derive(ToXml)]
struct Foo {
    bar: Bar,
}

struct Bar {
    flag: bool,
}

impl ToXml for Bar {
       fn serialize(.., field_data: ..) -> (...) {
                 field_data.unwrap().name = "Something"; // field_name passed from parent is "bar"
                 ...
       }
}
`
But I don't have a strong opinion about that, we can change it to immutable.

Reply:
Let's make it immutable for now, can always revisit the decision later.